File: elgcsharp/PrivateKey.cs
namespace sl.crypto.elgamal.keys
{
	using sl.crypto.elgamal.parameter;
	using sl.crypto.util;
	/**
	 * <summary>
 	 * Autor: Matthias Koch
 	 * Version: 0.5
 	 * 
 	 * Klasse welches ein konkretes Objekt für die Darstellung des privaten Schlüssels
 	 * darstellt.
 	 * </summary> 
 	 */
	public class PrivateKey : Key
	{
		private BigInteger x;
		//erlaube lesenden Zugriff
		public BigInteger X
		{
			get{return x;}
		}
		/**
		 * 
		 * <param name="x"/>
		 * <param name="params"/>
		 */
		public PrivateKey(BigInteger x, Parameter parameter):base(true, parameter)
		{			
			this.x = x;
		}
		/**
		 * 
		 */
		public bool equals(object obj)
		{
			if (!(obj is PrivateKey))
			{
				return false;
			}			
			PrivateKey pKey = (PrivateKey) obj;
			if (!pKey.X.Equals(x))
			{
				return false;
			}
			return Equals(obj);
		}
	}
}

File: elgjava/sl/crypto/elgamal/keys/PrivateKey.java
package sl.crypto.elgamal.keys;

import java.math.BigInteger;

import sl.crypto.elgamal.parameter.Parameter;

/**
 * Klasse welche ein konkretes Objekt für die Darstellung des privaten Schlüssels
 * darstellt.
 * 
 * @author Matthias Koch
 * @version 0.5 
 */
public class PrivateKey extends Key
{
	private BigInteger x;
	/**
	 * 
	 * @param x
	 * @param params
	 */
	public PrivateKey(BigInteger x, Parameter params)
	{
		super(true, params);
		this.x = x;
	}
	/**
	 * 
	 * @return BigInteger x
	 */
	public BigInteger getX()
	{
		return x;
	}
	/**
	 * 
	 */
	public boolean equals(Object obj)
	{
		if (!(obj instanceof PrivateKey))
		{
			return false;
		}
		PrivateKey pKey = (PrivateKey) obj;
		if (!pKey.getX().equals(x))
		{
			return false;
		}
		return super.equals(obj);
	}
}